src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -243,9 +246,37 @@ function _goderya(ybus::SparseArrays.SparseMatrixCSC)
     return I
 end
 
-function validate_connectivity(sys::System)
-    a = Adjacency(sys; check_connectivity = false)
-    return validate_connectivity(a.data, collect(get_components(Bus, sys)), a.lookup[1])
+function validate_connectivity(sys::System; method::Union{Nothing, Function} = nothing)
+    buses = sort(
```

Comment:
You are calling the same code for the adjacency and the validation. Can we move it to a separate function.